src/gsy_e/models/market/two_sided.py

```diff
@@ -291,6 +302,24 @@ def match_recommendations(
             recommended_pair = BidOfferMatch.from_dict(recommendations.pop(0))
             market_offer = self.offers.get(recommended_pair.offer["id"])
             market_bid = self.bids.get(recommended_pair.bid["id"])
+
+            # TODO: This is a temporary solution based on the fact that trading strategies do not
+            # post multiple bids or offers on the same market at the moment. Will be shortly
+            # replaced by a global offer / bid identifier instead of tracking the original order
+            # by seller / buyer.
+            if not market_offer:
+                market_offer = self._get_offer_from_seller_origin_id(
+                    recommended_pair.offer["seller_origin_id"])
+                if market_offer is None:
+                    raise InvalidBidOfferPairException("Offer does not exist in the market")
+                recommended_pair.offer = market_offer.serializable_dict()
+            if not market_bid:
+                market_bid = self._get_bid_from_buyer_origin_id(
+                    recommended_pair.bid["buyer_origin_id"])
+                if market_bid is None:
+                    raise InvalidBidOfferPairException("Bid does not exist in the market")
```

Comment:
@spyrostz do you believe this exception should break the simulation? I'm running some simulations and imo once the code reaches these conditions, we should `continue` rather than stopping the simulation

Reply:
I think that the caller code is not stopping the simulation IIRC (the external matcher class at least). I agree that the simulation should not stop due to that.